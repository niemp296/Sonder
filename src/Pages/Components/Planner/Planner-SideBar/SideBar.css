#planner-sidebar{
    padding: 1rem;
    justify-content: left;
    height: auto;
}
#planner-sidebar li{
    padding-bottom: 1rem;
}
#planner-sidebar li:hover{
    padding-bottom: 1rem;
    opacity:0.8;
}

div#planner-date-range {
    display: flex;
}
div#planner-days-icons {
    display: flex;
    flex-direction: column;
}