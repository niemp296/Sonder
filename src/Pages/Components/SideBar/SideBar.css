section.SideBar {
    padding: 1rem;
}