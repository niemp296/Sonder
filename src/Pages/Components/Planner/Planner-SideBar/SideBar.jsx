import React from 'react';
import './SideBar.css'
//import FlightTakeoffIcon from '@material-ui/icons/FlightTakeoff';
//import HotelIcon from '@material-ui/icons/Hotel';
import DateRangeIcon from '@material-ui/icons/DateRange';
import axios from 'axios'


//TODO: update database whenever the + or - is clicked

export default class SideBar extends React.Component {

    static propTypes = {
        //contains prop variables
    }
    constructor(props) {
        super(props);
        this.state = {
            author: "",
            budget: 0,
            locations: [],
            name: ""
        }
        this.populateState();
    }

    static defaultProps = {
        // contains default props
    }

    populateState = () =>{
        axios.get("http://localhost:5000/api/plans/" + this.props.plan_id)
        .then((response) =>{
            console.log(response);
            this.setState({
                author: response.data.author,
                budget: response.data.budget,
                locations: response.data.locations,
                name: response.data.name
            })
        })
    }

    renderPlannerDayButton(){
        const buttons =[];
        for(let i = 1; i<= this.state.locations.length; i++){
            buttons.push(React.createElement('button', { 
                className: "btn btn-light btn-outline-dark planner-day-icon mb-2",
                onClick: () => this.props.handleClick("Activity", i - 1)
            }, i));
        }
        return buttons;
    }

    incrementNumDays = () => {
        const loc = this.state.locations;
        loc.push({
            morning : [], afternoon : [], evening: []
        })
        
        const new_plan_data = {
            author: this.state.author,
            budget: this.state.budget,
            locations: loc,
            name: this.state.name
        }
        axios.put('http://localhost:5000/api/plans/' + this.props.plan_id, new_plan_data)
            .then((response) =>{
                this.setState({
                    locations: loc
                })
                console.log(this.state.locations);
            })
            .catch(error =>{
                console.log("error updating user data", error)
            })
    }

    decrementNumDays = () => {
        let c = window.confirm("Are you sure you want to delete day " + (this.state.locations.length).toString() + "?");
        if(c === true){
            this.props.handleClick("Activity", this.state.locations.length -2)
            const loc = this.state.locations;
            loc.pop();
            
            const new_plan_data = {
                author: this.state.author,
                budget: this.state.budget,
                locations: loc,
                name: this.state.name
            }
            axios.put('http://localhost:5000/api/plans/' + this.props.plan_id, new_plan_data)
                .then((response) =>{
                    this.setState({
                        locations: loc
                    })
                    console.log(this.state.locations);
                })
                .catch(error =>{
                    console.log("error updating user data", error)
                })
            } 
    }

    //TODO: make all symbols clickable, let user add days
    render() {
        return (
            <div className="btn-group-vertical" id="planner-sidebar">
                {/*}
                <button 
                className="mb-2 btn-outline-dark" 
                onClick = {() => this.props.handleClick("Stay")}
                >
                    <HotelIcon fontSize="large"/>
                </button>
                <button 
                className="mb-2 btn-outline-dark"
                onClick = {() => this.props.handleClick("Flight")}
                >
                    <FlightTakeoffIcon fontSize="large"/>
                </button>
                        */}
                <div id= "planner-date-range">
                    <button 
                    className="mb-2 btn-outline-dark" 
                    onClick={() => this.props.handleClick("Stay")}>
                        <DateRangeIcon fontSize="large"/>
                    </button>
                </div>
                <div id="planner-days-icons">
                    {this.renderPlannerDayButton()}
                    <button 
                    className="btn btn-light btn-outline-dark planner-day-icon mb-2"
                    onClick = {() => {
                        this.incrementNumDays();
                    }}
                    >+</button>
                     <button 
                    className="btn btn-light btn-outline-dark planner-day-icon mb-2"
                    onClick = {() => {
                        this.decrementNumDays();
                    }}
                    >-</button>
                </div>
            </div>
        );
    }
}