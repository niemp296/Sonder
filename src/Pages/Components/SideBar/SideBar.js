import React from 'react';
import SearchBar from '../SearchBar/SearchBar'
import './SideBar.css'
import list from '../../../TestData/search.json'

export default class SideBar extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
          list: list.entries
        };
      }
    render() {
        return (
            <section className="SideBar">               
                <SearchBar items={this.state.list}/>
            </section>
        );
    }
}