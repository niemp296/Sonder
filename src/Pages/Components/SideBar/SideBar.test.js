import React from 'react';
import { mount, shallow , configure} from 'enzyme';
import SideBar from './SideBar';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

describe("Test case for SideBar layout", () => {
    it("should render side bar page", () => {
        const wrapper = shallow(<SideBar />);
    });

    it("should render initial layout", () => {
        // when
        const component = shallow(<SideBar/>);
        // then
        expect(component.getElements()).toMatchSnapshot();
    });
});


