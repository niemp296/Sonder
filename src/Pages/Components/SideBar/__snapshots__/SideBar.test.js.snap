// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test case for SideBar layout should render initial layout 1`] = `
Array [
  <section
    className="SideBar"
  >
    <SearchBar />
  </section>,
]
`;
